File: objectModel/Python/cdm/persistence/cdmfolder/projections/operation_include_attributes_persistence.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

from typing import TYPE_CHECKING

from cdm.enums import CdmObjectType
from cdm.persistence.cdmfolder.types import OperationIncludeAttributes

from .operation_base_persistence import OperationBasePersistence

if TYPE_CHECKING:
    from cdm.objectmodel import CdmCorpusContext, CdmOperationIncludeAttributes
    from cdm.utilities import ResolveOptions, CopyOptions


class OperationIncludeAttributesPersistence:
    """Operation IncludeAttributes persistence"""

    @staticmethod
    def from_data(ctx: 'CdmCorpusContext', data: 'OperationIncludeAttributes') -> 'CdmOperationIncludeAttributes':
        if not data:
            return None

        include_attributes_op = OperationBasePersistence.from_data(ctx,
            CdmObjectType.OPERATION_INCLUDE_ATTRIBUTES_DEF, data)  # type: OperationIncludeAttributes
        include_attributes_op.include_attributes = data.includeAttributes

        return include_attributes_op

    @staticmethod
    def to_data(instance: 'CdmOperationIncludeAttributes', res_opt: 'ResolveOptions', options: 'CopyOptions') -> 'OperationIncludeAttributes':
        if not instance:
            return None

        obj = OperationBasePersistence.to_data(instance, res_opt, options)  # type: OperationIncludeAttributes
        obj.includeAttributes = instance.include_attributes

        return obj

File: objectModel/TypeScript/Persistence/CdmFolder/Projections/OperationIncludeAttributesPersistence.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import {
    CdmCorpusContext,
    cdmObjectType,
    CdmOperationIncludeAttributes,
    copyOptions,
    resolveOptions
} from '../../../internal';
import { OperationIncludeAttributes } from '../types';
import { OperationBasePersistence } from './OperationBasePersistence';

/**
 * Operation IncludeAttributes persistence
 */
export class OperationIncludeAttributesPersistence {
    public static fromData(ctx: CdmCorpusContext, object: OperationIncludeAttributes): CdmOperationIncludeAttributes {
        if (!object) {
            return undefined;
        }

        const includeAttributesOp: CdmOperationIncludeAttributes = OperationBasePersistence.fromData(ctx, cdmObjectType.operationIncludeAttributesDef, object);
        includeAttributesOp.includeAttributes = object.includeAttributes;

        return includeAttributesOp;
    }

    public static toData(instance: CdmOperationIncludeAttributes, resOpt: resolveOptions, options: copyOptions): OperationIncludeAttributes {
        if (!instance) {
            return undefined;
        }

        const data: OperationIncludeAttributes = OperationBasePersistence.toData(instance, resOpt, options);
        data.includeAttributes = instance.includeAttributes;

        return data;
    }
}
